src/Datasource/Paging/NumericPaginator.php

```diff
@@ -267,7 +267,7 @@ protected function getQuery(RepositoryInterface $object, ?QueryInterface $query,
 
         if ($query === null) {
             $args = [];
-            $type = !empty($options['finder']) ? $options['finder'] : 'all';
```

Comment:
I feel like !empty is the positive one here
So that rector rule is a bit counter productive.

Reply:
Its more about the unnessasary negation when you are analyzing the code

Reply:
It should also be about cognitive understanding - for the dev

Reply:
I like the unnecessary negation removal from all.